src/js/calculator.js

```diff
@@ -0,0 +1,130 @@
+import { ALERT_MESSAGE } from "./constants/messages.js";
+import { MAXIMUM_DIGITS_LENGTH, MODIFIER, OPERATION, DEFAULT_NUMBER } from "./constants/calculator.js";
+
+const operationExecutor = {
+  [OPERATION.ADDITION]: (leftValue, rightValue) => leftValue + rightValue,
+  [OPERATION.SUBTRACTION]: (leftValue, rightValue) => leftValue - rightValue,
+  [OPERATION.MULTIPLICATION]: (leftValue, rightValue) => leftValue * rightValue,
+  [OPERATION.DIVISION]: (leftValue, rightValue) =>
+    Math.floor(leftValue / rightValue),
+};
+
+const initialState = {
+  totalValue: DEFAULT_NUMBER,
+  leftValue: null,
+  rightValue: null,
+  operation: null,
+};
+
+class Calculator {
+  constructor({ $total, $digits, $operations, $modifiers }) {
+    this.$total = $total;
+    this.$digits = $digits;
+    this.$modifiers = $modifiers;
+    this.$operations = $operations;
+    this.state = {
+      ...initialState,
+    };
+  }
+
+  init() {
+    this.$digits.addEventListener("click", this.onDigitClick.bind(this));
+    this.$operations.addEventListener(
+      "click",
+      this.onOperationClick.bind(this)
+    );
+    this.$modifiers.addEventListener("click", this.onModifierClick.bind(this));
+  }
+
+  renderDisplay(value) {
+    this.$total.textContent = value;
+  }
+
+  calculate() {
+    const { leftValue, rightValue, operation, totalValue } = this.state;
+
+    if (operation === null) {
+      alert(ALERT_MESSAGE.NO_OPERATION);
+      
+      return;
+    }
+
+    if (rightValue === null) {
+      alert(ALERT_MESSAGE.NO_RIGHT_VALUE);
+
+      return;
+    }
+
+    const calculatedValue = operationExecutor[operation](leftValue ?? totalValue, rightValue);
+
+    this.state.totalValue = calculatedValue;
+    this.renderDisplay(calculatedValue);
+    this.resetOperationStates();
+  }
+
+  resetOperationStates() {
+    this.state = {
+      ...this.state,
+      leftValue: null,
+      rightValue: null,
+      operation: null,
+    };
+  }
+
+  onDigitClick({ target }) {
+    const currentValueKey =
+      this.state.operation === null ? "leftValue" : "rightValue";
+    const newValue =
+      (this.state[currentValueKey] ?? 0) * 10 + Number(target.dataset.digit);
```

Comment:
```suggestion
    const memorizedValue = (this.state[currentValueKey] ?? 0) * 10;
    const currentValue = memorizedValue + Number(target.dataset.digit);
```
변수선언이 늘어나긴하지만 가독성을위해 분리해보는 방향성을 생각해볼수도 있을거같아요.